adr/2022-11-stable-spec.md

```diff
@@ -0,0 +1,178 @@
+patch releases mid-cycle. Each release is a distinct version of JSON Schema and
+has no compatibility guarantees between versions.
+
+## Decision Outcome
+The decision is to go with Option 1 while leaving discussion open for aspects of
+Option 2 that could be adopted within the constraints of Option 1.
+
+Option 2 uses an immutable spec where each release replaces the last while
+Option 1 uses a mutable spec. The outcome of having only one current version of
+the spec is achieved with either option, but the mutable spec allows us to
+remove some unnecessary roadblocks in our development processes and allows us to
+release a stable spec much sooner.
+
+Option 2's restructuring of the spec into "Core Semantics" and "Standard
+Extensions" isn't specifically ruled out, but spec evolution is expected to be
+done primarily through mutation of the spec guided by the stability process
+rather than through extension. Option 1 puts no constraint on the structure of
+the spec and restructuring is allowed at any time as long as it doesn't break
+compatibility requirements.
+
+## Pros and Cons of the Options
+The biggest benefit is shared between Option 1 and Option 2. Both approaches
+result in a stable spec. This will have benefits for both implementers and
+users. Because of the compatibility requirements, whenever you write a schema,
+you will never need to change it just to keep up with changes to JSON Schema.
+This is also better for implementers because they don't have to maintain
+separate code with different semantics in different versions. They just need to
+code for the current release and they will automatically have support for past
+releases.
+
+### Option 1 - TC-39 Inspired
+The two things that make this option stand out are the stability model governing
+spec evolution and the mutability of the spec document.
+
+Having a mutable spec allows us to make clarifications and bug fixes immediately
+rather than having to wait months or years for the next release to go out. It
+also allows us to iterate faster on unstable features which would allow us to
+get them to a stable state much sooner. For example, we have changes to dynamic
+references that have been agreed upon and ready to go for over a year, but users
+can't benefit from the change until we can get the next full release published.
+With this model, the change could have been made available for over a year now
+and we would have a years worth of feedback on it's use. Having a mutable spec
+also allows us to introduce new features without having to wait for a release.
+For example, the `propertyDependencies` keyword has also been waiting for months
+for a release. Users could have been benefiting from it for months and providing
+feedback.
+
+The downside of a mutable spec is that it can be more difficult for implementers
+and users to track when changes happen. We will need to be better at
+communicating changes in blog posts or equivalent.
```

Comment:
I have some automation ideas on this.
Any such concept though would require us to track "latest modified" status of keywords in a structured data file, so automation can be run against such data.
This would add a burden to spec authors, unless unstable features/keywords are written in individual files, and merged into a single spec artifact at workflow time. In which case, changes would be easier to catch, but you still wouldn't know if it was a change requiring implementation change or not unless a human specified such.

Reply:
This is a bit out of scope for this PR, but I think I'd argue that this communication is better done by a person than an automation. An automation can't explain why something changed.

Reply:
Automation could provide reminders for one of us to post on a change, though.